source/initial_temperature/continental_geotherm.cc

```diff
@@ -0,0 +1,257 @@
+      AssertThrow(std::find(heating_models.begin(), heating_models.end(), "compositional heating") != heating_models.end(),
+                  ExcMessage("The continental geotherm initial temperature plugin requires the compositional heating plugin."));
+
+      // Check that the required material model ("visco plastic") is used
+      AssertThrow((dynamic_cast<MaterialModel::ViscoPlastic<dim> *> (const_cast<MaterialModel::Interface<dim> *>(&this->get_material_model()))) != 0,
+                  ExcMessage("The continental geotherm initial temperature plugin requires the viscoplastic material model plugin."));
+    }
+
+
+    template <int dim>
+    double
+    ContinentalGeotherm<dim>::
+    initial_temperature (const Point<dim> &position) const
+    {
+      const double depth = this->get_geometry_model().depth(position);
+
+      return temperature(depth);
+    }
+
+
+    template <int dim>
+    double
+    ContinentalGeotherm<dim>::
+    temperature (const double depth) const
+    {
+      // Compute some constants to calculate the temperatures T1 and T2 at the interfaces
+      // between the layers upper crust/lower crust and lower crust/lithospheric mantle.
+      const double a = 0.5*densities[0]*heat_productivities[0]*thicknesses[0] + 0.5*densities[1]*heat_productivities[1]*thicknesses[1] + conductivities[0]/thicknesses[0]*T0;
+      const double b = 1./(conductivities[0]/thicknesses[0]+conductivities[1]/thicknesses[1]);
+      const double c = 0.5*densities[1]*heat_productivities[1]*thicknesses[1] + conductivities[2]/thicknesses[2]*LAB_isotherm;
+      const double d = 1./(conductivities[1]/thicknesses[1]+conductivities[2]/thicknesses[2]);
+
+      //Temperature at boundary between layer 1 and 2
+      const double T1 = (a*b + conductivities[1]/thicknesses[1]*c*d*b) / (1.-(conductivities[1]*conductivities[1])/(thicknesses[1]*thicknesses[1])*d*b);
+      //Temperature at boundary between layer 2 and 3
+      const double T2 = (c + conductivities[1]/thicknesses[1]*T1) * d;
+
+      // Temperature in layer 1
+      if (depth <= thicknesses[0])
+        return -0.5*densities[0]*heat_productivities[0]/conductivities[0]*std::pow(depth,2) + (0.5*densities[0]*heat_productivities[0]*thicknesses[0]/conductivities[0] + (T1-T0)/thicknesses[0])*depth + T0;
+      // Temperature in layer 2
+      else if (depth <= thicknesses[0]+thicknesses[1])
+        return -0.5*densities[1]*heat_productivities[1]/conductivities[1]*std::pow(depth-thicknesses[0],2.) + (0.5*densities[1]*heat_productivities[1]*thicknesses[1]/conductivities[1] + (T2-T1)/thicknesses[1])*(depth-thicknesses[0]) + T1;
+      // Temperature in layer 3
+      else if (depth <= thicknesses[0]+thicknesses[1]+thicknesses[2])
+        return (LAB_isotherm-T2)/thicknesses[2] *(depth-thicknesses[0]-thicknesses[1]) + T2;
+      // Return a constant sublithospheric temperature of 10*LAB_isotherm.
```

Comment:
Will a factor  10 always be sufficient to exceed the adiabatic temperature? I imagine so, but perhaps 100 to be safe?